documentation/src/main/asciidoc/topics/con_protobuf_encoding.adoc

```diff
@@ -0,0 +1,52 @@
+[id='protobuf-encoding_{context}']
+= Protobuf cache encoding
+Protocol Buffers (Protobuf) is a lightweight binary media type for structured data.
+As a cache encoding, Protobuf gives you excellent performance as well as interoperability between client applications in different programming languages and remote server endpoints.
+
+The following example shows a Protobuf message that describes a `Person` object:
+
+[source,json,options="nowrap",subs=attributes+]
+----
+include::protobuf/personalshopper.proto[]
+----
+
+[discrete]
+== Interoperability
+
+Because it is language neutral, Protobuf encoding means {brandname} can handle requests from client applications written in Java, C{plusplus}, C#, Dart, Python, Go, and more.
+
+//Downstream content
+ifdef::downstream[]
+[TIP]
+====
+Always use Protobuf encoding for integration with any Red Hat technology because it allows communication between applications and services.
+====
+endif::downstream[]
+
+Protobuf also enables clients on different remote endpoints, Hot Rod or REST, to operate on the same data.
+Because it is backed by the REST API, you can access and work with Protobuf-encoded caches through {brandname} Console.
+
+[NOTE]
+====
+{brandname} uses a ProtoStream library to encode caches with the `application/x-protostream` media type.
+This encoding is similar to `application/x-protobuf` but the two media types have some differences that mean they are not interchangeable.
```

Comment:
Now here I have to ask the connaiseurs, why are x-protobuf and x-protostream different and not interchangeable?  I have no idea why we have invented x-protostream. x-protobuf whould have been enough. Any thoughts anyone ?

Reply:
@anistor it's never too late. maybe we can deprecate x-protostream in Infinispan 13 and start using x-protobuf instead?

Reply:
@anistor I have suggested this addition because AFAIK it's not possible to write a value with protobuf-java and read it with protostream (or vice-versa). Perhaps I was wrong, but they are still not interchangeable in the sense that we don't recognize `application/x-protobuf`.

Reply:
@danberindei and @anistor Do you guys feel like we need to make any changes here or is this OK? I think it's a good level of detail for the reader but maybe the wording is slightly negative? In Korea there's this great colloquialism "same same but different". That's kind of the spirit I'm aiming for here, not sure if it comes across.

Reply:
> I have suggested this addition because AFAIK it's not possible to write a value with protobuf-java and read it with protostream (or vice-versa).

Is this only the case if `WrappedMessage` has been used? I thought that if a user marshalls/unmarshalls data with Protostream via `ProtobufUtil.to/fromByteArray` that the two would be compatible.

Reply:
@danberindei and @ryanemerson TIL but I think we need to provide some more specific guidance on when to use those ProtoStream parameters with encodings. I can take note of this and will add it to the content plan for 13.

Reply:
Not sure if the type parameter in the protostream media type affects the conversion (or at least I don't remember seeing it)